akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -10,35 +10,34 @@ import akka.actor.Cancellable
 import akka.dispatch.ExecutionContexts
 import akka.event.Logging
 import akka.stream.FlowMonitorState._
-import akka.stream._
+import akka.stream.{ Shape, _ }
 import akka.stream.scaladsl._
 import akka.stream.impl.Stages.DefaultAttributes
 import akka.stream.stage._
 
 import scala.concurrent.{ Future, Promise }
 import scala.concurrent.duration.FiniteDuration
 import akka.stream.impl.StreamLayout._
-import akka.stream.impl.ReactiveStreamsCompliance
+import akka.stream.impl.{ LinearTraversalBuilder, ReactiveStreamsCompliance }
 
+import scala.annotation.unchecked.uncheckedVariance
 import scala.util.Try
 
 /**
  * INTERNAL API
  */
-final case class GraphStageModule(
-  shape:      Shape,
+// TODO: Fix variance issues
+final case class GraphStageModule[+S <: Shape @uncheckedVariance, +M](
```

Comment:
someone can maybe tell me why this does not work anymore. Or is this the right way to fix it.